whitepapers/0011_accountant.md

```diff
@@ -0,0 +1,190 @@
+
+In order for the accountant to have accurate values for the balances of the various accounts we will need to replay all transactions since the initial launch of the token bridge.  Similarly it may be necessary to backfill missing transfers if the accountant is disabled for any period of time.
+
+To handle both these issues, the accountant will have the ability to directly process tokenbridge transfer VAAs.  As long as the VAAs are signed by a quorum of guardians from the latest guardian set, the accountant will use them to update its on-chain state of token balances.
+
+Submitting VAAs will not disable any of the core checks though: a VAA that would cause the balance of an account to become negative will still be rejected even if it has a quorum of signatures.  This can be used to catch problematic transfers after the fact.
+
+### Handling Rejections
+
+Legitimate transactions should never be rejected so a rejection implies either a bug in the accountant or an actual attack on the bridge.  In both cases manual intervention is required.
+
+#### Alerting
+
+Whenever a guardian observes an error (either via an observation response or via an error event), it will increase a prometheus error counter.  Guardians will be encouraged to set up alerts whenever these metrics change so that they can be notified when an error occurs.
+
+## Deployment Plan
+
+- Deploy wormchain and have a quorum of guardians bring their nodes online.
+- Deploy the accountant contract to wormchain and backfill it with all the tokenbridge transfer VAAs that have been emitted since inception.
+- Enable the accountant in log-only mode.  In this mode, each guardian will submit its observations to the accountant but will not wait for the transfer to commit before signing the VAA.  This will give us some time to shake out any remaining issues with the implementation without impacting the network.
+- Once we have been running in log-only mode for some time and have built up confidence that it is operating correctly, switch to enforcing-mode.  In this mode, guardians will not sign VAAs for token transfers until the transfer is committed by the accountant.  As long as at least a quorum of guardians submit their observations to the accountant, we only need a super-minority (1/3) to turn on enforcing-mode for it to be effective.
+
+## Security Considerations
+
+### Threat Model
+
+Since the Global Accountant is implemented as an Integrity Checker, it inherits the threat model of Integrity Checkers. Therefore, it is only able to block messages, but it cannot create or modify messages.
+
+### Susceptibility to wormchain downtime
+
+Once all token transfers become gated on approval from the accountant, the uptime of the guardian network for those transfers will become dependent on the uptime of wormchain itself.  Any downtime for wormchain would halt all token transfers until the network is back online (or 2/3 of the guardians choose to disable the accountant).
```

Comment:
2/3+

Reply:
Fixed by @conorpp in e3f7a7d5d2ec5e7220c682b4a48b5e88eb40dd08.